src/index.c

```diff
@@ -734,8 +734,11 @@ void git_index_entry__init_from_stat(
 {
 	entry->ctime.seconds = (git_time_t)st->st_ctime;
 	entry->mtime.seconds = (git_time_t)st->st_mtime;
-	/* entry->mtime.nanoseconds = st->st_mtimensec; */
-	/* entry->ctime.nanoseconds = st->st_ctimensec; */
+#if !defined(GIT_WIN32) && !defined(__APPLE__)
+	/* Apple and Windows doesn't provide these struct stat fields. */
```

Comment:
Indeed.  Certainly `struct __stat64` on Windows has the `nanoseconds` field.  It would be nice if `USE_NSEC` Just Worked there.